scripts/server.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+'''
+An HTTP server that proxies requests to solr to search for similar incipits.
+
+Run a debug server for development with:
+    $ export FLASK_APP=scripts/server.py FLASK_ENV=development
+    $ flask run
+
+'''
+import os
+
+from flask import Flask, g, jsonify, render_template, request
+from parasolr.query import SolrQuerySet
+from parasolr.solr.client import SolrClient
+
+from scripts import __version__
+
+# read settings from environment on startup
+SOLR_URL = os.getenv('SOLR_URL', 'http://localhost:8983/solr/')
+SOLR_CORE = os.getenv('SOLR_CORE', 'pemm')
+
+# create a new flask app from this module
+app = Flask(__name__)
+
+
+@app.route('/')
+def root():
+    '''Display version info and a basic html search form.'''
+    return render_template('index.html', version=__version__)
+
+
+@app.route('/search', methods=['GET', 'POST'])
+def search():
+    '''Search for an incipit and return a list of matching results.'''
+    if request.method == 'POST':
+        search_term = request.form['incipit']
+        output_format = request.form.get('format', '')
+    elif request.method == 'GET':
+        search_term = request.args.get('incipit', '')
+        output_format = request.args.get('format', '')
```

Comment:
could `output_format = request.args.get('format', '')` be outside this conditional?

Reply:
From my brief reading of the flask docs we need to use `form` for POST and `args` for GET. If there's a way to simplify this, I'm all for it. (Maybe just determine which request object to use and then it's same logic once we have?)

Reply:
ah, knew i was missing something. it's fine for now then imo; we could revisit it we add more logic specific to the request type